.circleci/config.yml

```diff
@@ -219,22 +220,26 @@ commands:
 
   win_build:
     description: Run build on Windows
+    parameters: 
+      python_version: 
+        type: string
     steps:
       - checkout
-      - restore_cache:
-          keys:
-            - v0-{{ arch }}-dependencies-{{ checksum "package.json" }}-{{ checksum "package/requirements.txt" }}-{{ checksum "package/test_requirements.txt" }}
-            # fallback to using the latest cache if no exact match is found
-            - v0-{{ arch }}-dependencies-
-      - win_setup_conda
```

Comment:
just out of curiosity, I wonder what was the reason behind the these steps failing?

Reply:
it's to do with some caching on python venv. it is a bit confusing to me too :)

Reply:
Basically the caches currently aren't specific to the version of Python. e.g. when we try running the 3.10 job, the cache for 3.8 might be loaded up. The best way to ensure we have a clean testing environment is just to not cache anything at all so there's no interference. We can then _maybe_ add it back in in the future CI refactor ticket. If we do so then we should include the python version in the cache key so there's a different cache for each Python version though.

The reason I say _maybe_ is that saving and restoring cache on CircleCI is actually pretty slow, so sometimes there's not actually any time saving to be made compared to not caching and instead just making a fresh environment every time.